### ISSUE TITLE
quadrangular polygon object detection

### ISSUE THREAD

**Issue by armine105** [Deci-AI/super-gradients#1980]
### 💡 Your Question

Hello
How are you?
I want to detect a quadrangular polygon object such as a pool table below.

![image](https://github.com/Deci-AI/super-gradients/assets/135291100/389ca9e5-b620-461c-8568-4d105aa998ef)

Is it possible to do it using yolo-nas?
The "Polygon-Yolov5" project (https://github.com/XinzeLee/PolygonObjectDetection) supports this.

**Comment by BloodAxe** [comment]
I think you can achieve desired goal with YoloNAS-Pose model. If you consider a pool table corners as individual keypoints you can train pose estimation model with 4 keypoints corresponding to TL, TR, BR, BL corners. And from there you can reconstruct your polygon.

Please check https://github.com/Deci-AI/super-gradients/blob/master/notebooks/YoloNAS_Pose_Fine_Tuning_Animals_Pose_Dataset.ipynb notebook to learn more about fine-tuning YoloNAS-Pose model.

**Comment by armine105** [comment]
Thank you @BloodAxe
